compiler/AST/build.cpp

```diff
@@ -3130,3 +3133,15 @@ Expr* tryBangExpr(Expr* e)
 {
   return new CallExpr(PRIM_TRYBANG_EXPR, e);
 }
+
+Expr* convertAssignmentAndWarn(Expr* a, const char* op, Expr* b)
+{
+  if (0 == strcmp("=", op)) {
+    USR_FATAL_CONT(a, "Use == to check for equality in a conditional");
+  } else {
+    USR_FATAL_CONT(a, "Invalid conditional %s", op);
```

Comment:
Makes me wonder what Chapel code can lead to this case. Could you discuss this in a comment here (ok if briefly) or add a test?

Reply:
if a += b then
I'll add a test